docs: improve execution modes (#35412)
microsoft/playwright: docs/src/test-api/class-test.md

  test('runs in parallel 2', async ({ page }) => {});
  ```

> * Running tests in order, retry is per failed test. If one test fails, subsequent tests are still runned.

Reviewer: I'd not mention the "continue on failure" behavior - that's expected. It is only important to mention that for serial mode, and avoid unnecessary details for the default mode.

```suggestion
* Running tests in order, retrying each failed test independetly.
```

  This is the default mode. It can be useful to set it explicitly to override project configuration that uses `fullyParallel`.

  ```js

>   // All tests run in order, second test will run even if the first one fails.

Reviewer: ```suggestion
  // Tests in this file run in order. Retries, if any, run independently.
```
